java/daikon/dcomp/BuildJDK.java

```diff
@@ -3,375 +3,500 @@
+ * uninstrumented version of a method.
  */
 public class BuildJDK {
 
-  @Option(
-      "Instrument the given classfiles from the specified source directory (by default, src must be a jar file)")
-  public static boolean classfiles = false;
-
-  /** Synopsis for the dcomp command line. */
-  public static final String synopsis =
-      "daikon.BuildJDK [options] src dest [class_prefix | classfiles...]";
-
   /**
-   * Given an explicit {@code rt.jar} filename, or a root JDK or JRE directory, finds the {@code
-   * rt.jar} file.
+   * The "java.home" system property. Note that there is also a JAVA_HOME variable that contains
+   * {@code System.getenv("JAVA_HOME")}.
    */
-  public static class RtJarFinder {
-    private final String arg;
-
-    public RtJarFinder(String arg) {
-      super();
-      this.arg = arg;
-    }
+  public static final String java_home = System.getProperty("java.home");
 
-    /**
-     * @param spaceSeparatedJarLocations a list of potential jar filenames to look for, in case arg
-     *     was a directory. For example, if arg was /usr/bin/jd2sdk, and spaceSeparatedJarLocations
-     *     was "jre/lib/rt.jar", this could return /usr/bin/jd2sdk/jre/lib/rt.jar, if it existed.
-     */
-    public String findRtJar(String spaceSeparatedJarLocations) {
-      String jarFilename = arg;
-      if (!arg.endsWith("jar")) {
-        for (String candidate : spaceSeparatedJarLocations.split(" ")) {
-          String rt = jarFilename + candidate;
-          if (exists(rt)) {
-            return rt;
-          }
-        }
-      }
-      return jarFilename;
-    }
+  /**
+   * Instrument the class files given on the command line instead of the ones in the Java runtime
+   * library.
+   */
+  @Option("Instrument only the class files given on the command line")
+  public static boolean classfiles = false;
 
-    /** @return true iff libRt exists in the filesystem */
-    protected boolean exists(String libRt) {
-      return new File(libRt).exists();
-    }
-  }
+  /**
+   * Used when debugging to terminate processing if an error occurs. This flag is tested in
+   * DCInstrument.
+   */
+  @Option("Halt if an instrumentation error occurs")
+  public static boolean quit_if_error = false;
 
+  /** Print information about the classes being instrumented. */
   private static boolean verbose = false;
 
-  /** Whether or not to instrument java.lang.Object */
-  private static boolean skip_object = true;
-
+  /** Number of class files processed; used for progress display. */
   private int _numFilesProcessed = 0;
 
-  private static String static_map_fname = "dcomp_jdk_static_map";
+  /** Name of entry in output jar containing the statics map. */
+  private static String static_field_id_fname = "dcomp_jdk_static_field_id";
 
+  /** Names of all the classes processed. */
   private static List<String> all_classes = new ArrayList<>();
 
+  /** Names of all methods that DCInstrument could not process. Should be empty. */
```

Comment:
Are these simple names, or do they include class names, and if so, what is the format?